src/reduce.js

```diff
@@ -6,6 +6,40 @@ import {abs, exp, log} from './unary.js';
 import {sizeOfShape, Scalar, Tensor} from './lib/tensor.js';
 import {validateReduceParams} from './lib/validate-input.js';
 
+export function selectValuesToReduce(input, axes, inputLocation) {
+  validateReduceParams(input, {axes});
+
+  const outputShape = input.shape.slice();
+  for (let i = 0; i < axes.length; ++i) {
+    outputShape[axes[i]] = 1;
+  }
+
+  // Calculate the "strides" across the reduction dimensions given in axes.
+  axes.sort((a, b) => a - b);
+  const reduceDims = axes.map((axis) => input.shape[axis]);
+  const reduceElements = sizeOfShape(reduceDims);
+  const reduceStrides = new Array(axes.length);
+  reduceStrides[reduceStrides.length - 1] = 1;
+  for (let i = reduceStrides.length - 2; i >= 0; --i) {
+    reduceStrides[i] = reduceStrides[i + 1] * reduceDims[i + 1];
+  }
+
+  const valuesToReduce = [];
+  // Find all values to reduce.
+  for (let reduceIndex = 0; reduceIndex < reduceElements; ++reduceIndex) {
```

Comment:
How about `reducedElementCount`? `reduceElements` sounds like it's actually the elements themselves (the values).

Reply:
Updated.